src/main/webapp/script.js

```diff
@@ -15,10 +15,14 @@
 /*
  * Displays navbar authentication buttons according to login status.
  */
-window.onload = function fetchLogin() {
+function fetchLogin() {
   fetch('/login').then(response => response.json()).then(user => {
     // If user is logged in, show logout button in navbar.
     if (user.loggedIn == true) {
+      // If logged in user is not registered, redirect to signup page.
+      if(user.isUserRegistered == false) {
```

Comment:
Since this is a boolean I'm pretty sure you can just do something like this:

```suggestion
      if (!user.isUserRegistered) {
```

Reply:
Done!